airflow/jobs.py

```diff
@@ -968,6 +949,43 @@ def _process_task_instances(self, dag, queue, session=None):
                     self.log.debug('Queuing task: %s', ti)
                     queue.append(ti.key)
 
+    @provide_session
+    def _remove_zombies_from_executor(self,
```

Comment:
This new method is one side of a two-process information exchange.  This is called by the dag file loading process, and sets the state to ZOMBIED.  The "real" scheduler process will pick up task instances in this state, and remove them from its in-memory executor "running" and "queued" maps, which will allow the task to retry.